### PR TITLE
gnuradio: add codec2 dependency

### PR DESCRIPTION
**Pull request by rheyns**
Without this dependency building with `brew install --build-from-source gnuradio` fails when run on macOS Ventura 13.3.1 (a) (22E772610a)

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/gnuradio.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@ class Gnuradio < Formula
   url "https://github.com/gnuradio/gnuradio/archive/refs/tags/v3.10.6.0.tar.gz"
   sha256 "c1a8c09f975d224b27dad9c8e2e7fed1f7cde9bc04d30b004afe8e28bb3f45c6"
   license "GPL-3.0-or-later"
-  revision 1
+  revision 2
```
Comment (carlocab) [inline Formula/gnuradio.rb:9]: Let's not force everyone to reinstall this just to change a build dependency, please.
```suggestion
  revision 1
```

**Review by carlocab — CHANGES_REQUESTED** [review]
Thanks for the PR.

> Without this dependency building with `brew install --build-from-source gnuradio` fails when run on macOS Ventura 13.3.1 (a) (22E772610a)

We should understand why that's happening first, because I don't think this happens for the same macOS version in CI.

**Comment by rheyns** [comment]
> Thanks for the PR.
> 
> > Without this dependency building with `brew install --build-from-source gnuradio` fails when run on macOS Ventura 13.3.1 (a) (22E772610a)
> 
> We should understand why that's happening first, because I don't think this happens for the same macOS version in CI.

I can open an issue if that's a better venue for discussing the local build failure. The build failure was blocking me from what I actually wanted to do which is add the gr-iio module, so I figured I needed to address that first.

**Comment by carlocab** [comment]
We can discuss this here.

**Comment by rheyns** [comment]
Thanks for the feedback @carlocab, any hints on how to look up the last successful CI run for this formula? What the CI runs seems to be different from what the docs request for opening a PR so it seems like a good place to start looking for differences in build behavior.

**Comment by rheyns** [comment]
> We can discuss this here.

Great, one sec while I rerun without the codec2 dependency.

**Comment by rheyns** [comment]
So here's what I'm getting locally without explicitly adding `codec2` as a dependency:
https://gist.github.com/rheyns/25bb72733be82df4a304485a25b3e26b

**Comment by carlocab** [comment]
Will try to track down some build logs. You can also upload more comprehensive logs with
```
brew gist-logs gnuradio
```
Please also try uninstalling `codec2` (or, if that risks breaking too many things for you, `brew unlink codec2`) and see if you still get the build error.

**Comment by rheyns** [comment]
Only need `codec2` for GNU Radio so no problems uninstalling it.

I reproduced the failure by uninstalling gnuradio and now-unneeded dependencies including codec2 and then rebuilding locally from homebred-core master branch. Logs are here https://gist.github.com/rheyns/231e327465f4d903f0628f9a6e2eb86c and I've also attached the terminal session
[2023-06-06-gnuradio-install-no-codec2.txt](https://github.com/Homebrew/homebrew-core/files/11667203/2023-06-06-gnuradio-install-no-codec2.txt)

**Comment by carlocab** [comment]
I just dispatched a build in CI, and it looks like the build succeeded. It errored out anyway because it refused to overwrite the bottle that was already there.

**Comment by carlocab** [comment]
I noticed this flag in your gist logs: `-isystem /opt/local/include/codec2`

Do you have MacPorts installed, or `codec2` otherwise installed in `/opt/local`? I suspect that that's what's causing your build failure. Please uninstall `codec2` from there to fix this.

**Pull request closed without merging**